util/types/json/compare.go

```diff
@@ -114,7 +114,8 @@ func CompareJSON(j1 JSON, j2 JSON) (cmp int, err error) {
 			s2 := Serialize(j2)
 			cmp = bytes.Compare(s1, s2)
 		default:
-			cmp = 0
+			err = errors.Errorf(unknownTypeCodeErrorMsg, j1.typeCode)
```

Comment:
It will never happen because before here, on each JSON objects we call `Type` which will panic if the `typeCode` is invalid.